activestorage/lib/active_storage/service/s3_service.rb

```diff
@@ -95,6 +95,16 @@ def url_for_direct_upload(key, expires_in:, content_type:, content_length:, chec
       end
     end
 
+    def public_url(key, **)
```

Comment:
Both [GCS](https://github.com/googleapis/google-cloud-ruby/blob/afe59de2d33f5a1ad1c09c3c626337323ffe1c27/google-cloud-storage/lib/google/cloud/storage/file.rb#L1381) and [S3](https://github.com/aws/aws-sdk-ruby/blob/ef1cc2c3b667d6d746b0e3ad59087461be24d438/gems/aws-sdk-s3/lib/aws-sdk-s3/customizations/object.rb#L210) appear to have not have support for `content_type`, `disposition`, and `filename` options when generating public URLs.